modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/DatasetDisplay.js

```diff
@@ -340,7 +351,7 @@ function DatasetDisplay({
 						value={selectedItemsValue.join(',')}
 					/>
 				)}
-				{items?.length ?? 0 ? (
+				{(items?.length ?? 0) || overrideEmptyResultView ? (
```

Comment:
Seeing as this is complicated, can you just verify that this is indeed the desired behavior?

- if `items` is `undefined` or `items.length` is `0` → show `<EmptyResultMessage>`.
- if `items.length` is positive → show `<CurrentViewComponent>`.
- if `items.length` is `0` _but_ `overrideEmptyResultView` is `true` → `<CurrentViewComponent>`.
- if `items` is `undefined` _but_ `overrideEmptyResultView` is `true` → `<CurrentViewComponent>` (this one worries me).

Are we suse that `<CurrentViewComponent>` will gracefully handle `items` being `undefined`?

My suspicion is that this should perhaps be:

```
(items?.length) || (overrideEmptyResultView && items) ? (...) : (...)
```

or:

```
items={items || []}
```

Both of which are pretty horrible, but we're already inside a ternary at this point and that makes it hard to hoisting it up higher.

```js
```

Reply:
> Are we sure that <CurrentViewComponent> will gracefully handle items being undefined?

Yeah, I lack the usage cases to test all logical branches previously implemented by Commerce properly. Looking at the [code from Commerce](https://github.com/liferay/liferay-portal-ee/blob/7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/dataset_display/DatasetDisplay.js#L334) we can assume that views should indeed be able to handle `undefined` items.

And the only usage of `overrideEmptyResultView` I could find was [this](https://github.com/liferay/liferay-portal-ee/blob/7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/Wrapper.js#L35). And from there we can see that the view who receives the items is [this](https://github.com/liferay/liferay-portal-ee/blob/7.1.x/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemsList.js), which has its own empty results message handling logic. So that makes me believe they did it intentionally so that custom views could have custom empty results messages.